tests/cmdline/params/types/test_group.py

```diff
@@ -7,81 +7,140 @@
-
-    def test_get_by_uuid(self):
-        """
-        Verify that using the UUID will retrieve the correct entity
-        """
-        identifier = '{}'.format(self.entity_01.uuid)
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-    def test_get_by_label(self):
-        """
-        Verify that using the LABEL will retrieve the correct entity
-        """
-        identifier = '{}'.format(self.entity_01.label)
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-    def test_ambiguous_label_pk(self):
-        """
-        Situation: LABEL of entity_02 is exactly equal to ID of entity_01
-
-        Verify that using an ambiguous identifier gives precedence to the ID interpretation
-        Appending the special ambiguity breaker character will force the identifier to be treated as a LABEL
-        """
-        identifier = '{}'.format(self.entity_02.label)
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-        identifier = '{}{}'.format(self.entity_02.label, OrmEntityLoader.label_ambiguity_breaker)
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_02.uuid)
-
-    def test_ambiguous_label_uuid(self):
-        """
-        Situation: LABEL of entity_03 is exactly equal to UUID of entity_01
-
-        Verify that using an ambiguous identifier gives precedence to the UUID interpretation
-        Appending the special ambiguity breaker character will force the identifier to be treated as a LABEL
-        """
-        identifier = '{}'.format(self.entity_03.label)
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-        identifier = '{}{}'.format(self.entity_03.label, OrmEntityLoader.label_ambiguity_breaker)
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_03.uuid)
+@pytest.fixture
+def setup_groups(clear_database_before_test):
+    """Create some groups to test the `GroupParamType` parameter type for the command line infrastructure.
+
+    We create an initial group with a random name and then on purpose create two groups with a name that matches exactly
+    the ID and UUID, respectively, of the first one. This allows us to test the rules implemented to solve ambiguities
+    that arise when determing the identifier type.
+    """
+    entity_01 = Group(label='group_01').store()
+    entity_02 = AutoGroup(label=str(entity_01.pk)).store()
+    entity_03 = ImportGroup(label=str(entity_01.uuid)).store()
+    return entity_01, entity_02, entity_03
+
+
+def test_get_by_id(setup_groups):
+    """Verify that using the ID will retrieve the correct entity."""
+    entity_01, entity_02, entity_03 = setup_groups
+    param = GroupParamType()
```

Comment:
Same comment as previously, concerning perhaps creating a file-wide `PARAM` variable or a pytest fixture instead of having to repeat the same line for all tests?

Reply:
Done